includes/base/controls-stack.php

```diff
@@ -849,6 +849,15 @@ final public function add_responsive_control( $id, array $args, $options = [] )
 		$is_frontend_available = ! empty( $control_to_check['frontend_available'] );
 		$has_prefix_class = ! empty( $control_to_check['prefix_class'] );
 
+		$enable_duplication_mode = [
+			'loop-item',
+			'common',
+			'container',
+		];
+		if ( in_array( $this->get_name(), $enable_duplication_mode, true ) ) {
+			$responsive_duplication_mode = 'on';
```

Comment:
We can't do this solution, managing a list of elements here is not very maintainable and as @StyleShit mentioned above, currently contains pro widgets.

The solution has to do with the Responsive Duplication mode being `'off'` by default. It is off by default to make sure responsive controls are duplicated (controls added per devices) in a regular front-end load, where they are not necessary. This is for performance with additional breakpoints.

Each controls stack (elements, widgets) is only registered *once* and cached. Meaning, the first time a Container is rendered, its controls are registered and cached. The next container rendered on the page will get its controls from the cached stack.

Because of this, I see two potential solutions for this problem:
1. Intervene in the first time element/widget controls are registered and make sure that the responsive duplication mode is on when that happens, if a CSS file needs to be registered in that specific page load (details below).
2. If a CSS file is registered, we reset the stack cache completely, so all controls for all elements would have to be re-registered, and we set the duplication mode to 'on' before running the CSS generation process. This solution is less ideal because it means all of the controls are registered multiple times, which significantly affects on performance.

I believe the first solution is better because of the performance issue. I would do the following: 
1. On page load request, before any controls are registered, we check if the current page has any templates (The check itself would have to be done in Pro, with an action or filter to be triggered in Core).
3. If it does have templates (header, footer, loop etc.), we check any of them needs to generate a CSS file.
4. If at least one CSS file needs to be generated, we set the control duplication mode to `'on'` using `Plugin::$instance->breakpoints->set_responsive_control_duplication_mode( 'on' )`.

This way, if on a certain page load, any templates are missing their CSS files, the duplication mode will be set to on and responsive controls will be duplicated when they are added to the cached Controls Stack
@maork-elementor

Reply:
Thanks, @udidol, for the expanded detailed!

While I believe that the first solution is better, we cannot determine what will be displayed at the bottom of the page during page load as it may be dynamic or nested within a template. 

Therefore, I think we should proceed with the second solution.

Reply:
I understand. I tend to agree.

From looking into it with Maor, I believe the best way to do it would be here:
https://github.com/elementor/elementor/blob/1e8b85a6d3e3c8cbd53b8eca7cc7e5b8b4aeadad/core/files/css/base.php#L646-L671
Suggestion for a way to go:
1. We create a `delete_stack()`/`delete_all_stacks()` method in the `Controls_Manager` class which sets the `stacks` property to an empty array, deleting all cached stacks.

2. In the `parse_content()` method, where the controls are actually used for CSS generation, we check if responsive controls are already registered as duplicated (part of this solution would include finding a simple way to check this).

3. If the stacks are not already duplicated, delete all stacks.

4. Part of the solution would have to include a way to cache the deletion of the stacks, with a flag of some kind. This is necessary because re-registering all controls is a very expensive operation, and it only needs to happen once per a page load's code execution. If multiple CSS files are generated in one page load, the controls have already been re-registered fully duplicated, so there's no reason to do it again in the same execution run.

5. That's it, we should be golden.